src/invariant/ConservationOfLumens.cpp

```diff
@@ -35,6 +41,34 @@ calculateDeltaBalance(LedgerEntry const* current, LedgerEntry const* previous)
         return ((current ? current->data.claimableBalance().amount : 0) -
                 (previous ? previous->data.claimableBalance().amount : 0));
     }
+    case LIQUIDITY_POOL:
+    {
+        auto const* currentBody =
+            current ? &current->data.liquidityPool().body.constantProduct()
+                    : nullptr;
+        auto const* previousBody =
+            previous ? &previous->data.liquidityPool().body.constantProduct()
+                     : nullptr;
+
+        auto const& assetA =
+            (currentBody ? currentBody : previousBody)->params.assetA;
+        auto const& assetB =
+            (currentBody ? currentBody : previousBody)->params.assetB;
+
+        // At most one asset can be native
```

Comment:
Should this assumption be made here? We'll obviously have much bigger problems if this is somehow true, but I think this invariant should account for all lumen changes on the network.

Reply:
Good point, better safe than sorry. I made your change. But as you note, it'd be bad in many other ways if both were native.

Reply:
Your update doesn't change anything here. The `assetA` block returns, so you won't hit the `assetB` block if they're both native.

Reply:
Fixed.